jwql/instrument_monitors/common_monitors/cosmic_ray_monitor.py

```diff
@@ -0,0 +1,608 @@
+                uncal_file = os.path.join(self.obs_dir, os.path.basename(file_name))
+
+                try:
+                    pipeline_tools.calwebb_detector1_save_jump(uncal_file, self.obs_dir, ramp_fit=True, save_fitopt=False)
+                except:
+                    logging.info('Failed to complete pipeline steps on {}.'.format(uncal_file))
+                    pass
+
+                # Next we analyze the cosmic rays in the new data
+                for output_file in os.listdir(self.obs_dir):
+
+                    if 'jump' in output_file:
+                        jump_file = os.path.join(self.obs_dir, output_file)
+
+                    if self.nints == 1:
+                        if '0_ramp_fit' in output_file:
+                            rate_file = os.path.join(self.obs_dir, output_file)
+
+                    elif self.nints > 1:
+                        if '1_ramp_fit' in output_file:
+                            rate_file = os.path.join(self.obs_dir, output_file)
+
+                try:
+                    jump_head, jump_data, jump_dq = self.get_jump_data(jump_file)
+                except:
+                    logging.info('Could not open jump file: {}'.format(jump_file))
+
+                try:
+                    rate_data = self.get_rate_data(rate_file)
+                except:
+                    logging.info('Could not open rate file: {}'.format(rate_file))
+
+                jump_locs = self.get_jump_locs(jump_dq)
+                jump_locs_pre = self.group_before(jump_locs)
+
+                eff_time = jump_head['EFFEXPTM']
+
+                # Get observation time info
+                obs_start_time = jump_head['EXPSTART']
+                obs_end_time = jump_head['EXPEND']
+                start_time = julian.from_jd(obs_start_time, fmt='mjd')
+                end_time = julian.from_jd(obs_end_time, fmt='mjd')
+
+                cosmic_ray_num = len(jump_locs)
+                cosmic_ray_mags = self.get_cr_mags(jump_locs, jump_locs_pre, rate_data, jump_data, jump_head)
+
+                # Insert new data into database
+                try:
+                    cosmic_ray_db_entry = {'entry_date': datetime.datetime.now(),
+                                           'aperture': self.aperture,
+                                           'source_file': file_name,
+                                           'obs_start_time': start_time,
+                                           'obs_end_time': end_time,
+                                           'jump_count': cosmic_ray_num,
+                                           'magnitude': cosmic_ray_mags
+                                           }
+                    self.stats_table.__table__.insert().execute(cosmic_ray_db_entry)
+
+                    logging.info("Successfully inserted into database. \n")
+                except:
+                    logging.info("Could not insert entry into database. \n")
+
```

Comment:
It would be helpful here to put a few lines of code in order to delete the data files that were used (uncal, jump, rate, rateits, etc). This will help save disk space on the server. As commissioning has gone on, we've seen the leftover products from the other monitors can quickly fill up our available space.

Reply:
@bhilbert4 are we going to implement these changes?